apps/settings/js/modules/bluetooth/bluetooth_cod_mapper.js

```diff
@@ -0,0 +1,161 @@
+      17: 'audio-card',          // (Reserved)
+      18: 'audio-card'           // Gameing/Toy
+    },
+
+    Peripheral: {
+      1: 'input-gaming',         // Joystick
+      2: 'input-gaming',         // Gamepad
+      16: 'input-keyboard',      // Keyboard
+      17: 'input-keyboard',      // Keyboard
+      18: 'input-keyboard',      // Keyboard
+      19: 'input-keyboard',      // Keyboard
+      20: 'input-keyboard',      // Keyboard
+      21: 'input-keyboard',      // Keyboard
+      22: 'input-keyboard',      // Keyboard
+      23: 'input-keyboard',      // Keyboard
+      24: 'input-keyboard',      // Keyboard
+      25: 'input-keyboard',      // Keyboard
+      32: 'input-mouse',         // Pointing device
+      33: 'input-mouse',         // Pointing device
+      34: 'input-mouse',         // Pointing device
+      35: 'input-mouse',         // Pointing device
+      36: 'input-mouse',         // Pointing device
+      37: 'input-tablet',        // Digitizer tablet
+      38: 'input-mouse',         // Pointing device
+      39: 'input-mouse',         // Pointing device
+      40: 'input-mouse',         // Pointing device
+      41: 'input-mouse'          // Pointing device
+    },
+
+    Imaging: {
+      8: 'camera-photo',         // Camera, bit: XX1XXX
+      16: 'scanner',             // Scanner, bit: X1XXXX
+      32: 'printer'              // Printer, bit: 1XXXXX
+    }
+  };
+
+  var MajorDeviceClasses = {
+    1: MinorDeviceClasses.Computer,
+    2: MinorDeviceClasses.Phone,
+    3: 'network-wireless',       // LAN/Network Access Point Major Class
+    4: MinorDeviceClasses.AudioVideo,
+    5: MinorDeviceClasses.Peripheral,
+    6: MinorDeviceClasses.Imaging
+  };
+
+  var BluetoothClassOfDeviceMapper = {
+    /**
+     * The string indicating which kind of icon could be used to represent
+     * the device.
+     *
+     * @access public
+     * @memberOf BluetoothClassOfDeviceMapper
+     * @param {Object BluetoothClassOfDevice} cod
+     * @return {String}
+     */
+    getDeviceType: function btcodm_getDeviceType(cod) {
+      Debug('cod.majorDeviceClass = ' + cod.majorDeviceClass);
+      Debug('cod.majorServiceClass = ' + cod.majorServiceClass);
+      Debug('cod.minorDeviceClass = ' + cod.minorDeviceClass);
+      // Given an empty string to be default type.
+      var deviceType = '';
+      var majorDeviceClass = MajorDeviceClasses[cod.majorDeviceClass];
+      Debug('--> majorDeviceClass = ' + majorDeviceClass);
+      Debug('--> typeof majorDeviceClass = ' + typeof(majorDeviceClass));
+      if (typeof(majorDeviceClass) === 'string') {
+        // drop in LAN/Network Access Point Major Class
+        deviceType = majorDeviceClass;
+        Debug('--> major class return type = ' + deviceType);
+        return deviceType;
+      } else if (typeof(majorDeviceClass) === 'object') {
+        // drop in other Major Class
+        if (majorDeviceClass[cod.minorDeviceClass]) {
```

Comment:
Suggest to move this case as the first one because this will be the most hit case.